tests/test_async_code_match.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pathlib
+import ast
+import typing
+import re
+
+from absl.testing import absltest
+from absl.testing import parameterized
+
+WORKING_DIRS = [
+    (pathlib.Path(__file__).parent.parent / "google"),
+]
+EXEMPT_DIRS = ["notebook"]
+EXEMPT_DECORATORS = ["overload", "property", "setter", "abstractmethod", "staticmethod"]
+EXEMPT_FILES = ["client.py", "version.py", "discuss.py", "files.py"]
```

Comment:
Does the test still pass if these lists are empty?

Reply:
no!

So for files.py, we don't support async implementation, so it fails there.
For discuss.py, there are many inconsistencies with error messages and decorators, (so basically test fails as the code is not exactly the same/consistent)